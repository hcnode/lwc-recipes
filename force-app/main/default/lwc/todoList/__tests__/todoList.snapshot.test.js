import { createElement } from 'lwc';
import TodoList from 'c/todoList';

let TODOS = [
    { id: 1, description: 'Explore recipes', priority: false },
    { id: 2, description: 'Install Ebikes sample app', priority: false }
];

describe('c-todo-list', () => {
    afterEach(() => {
        // The jsdom instance is shared across test cases in a single file so reset the DOM
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('renders without any list items as default', () => {
        // Create initial element
        const element = createElement('c-todo-list', {
            is: TodoList
        });
        document.body.appendChild(element);

        expect(element).toMatchSnapshot();
    });

    it('renders multiple list items', () => {
        // Create initial element
        const element = createElement('c-todo-list', {
            is: TodoList
        });
        // Set public properties
        element.todos = TODOS;
        document.body.appendChild(element);

        expect(element).toMatchSnapshot();
    });

    it('is accessible when todo items added', () => {
        // Create initial element
        const element = createElement('c-todo-list', {
            is: TodoList
        });
        // Set public properties
        element.todos = TODOS;
        document.body.appendChild(element);

        return Promise.resolve().then(() => expect(element).toBeAccessible());
    });
});
