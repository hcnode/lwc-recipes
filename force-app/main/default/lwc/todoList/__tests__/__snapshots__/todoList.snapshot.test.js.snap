// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-todo-list renders multiple list items 1`] = `
<c-todo-list>
  #shadow-root(open)
    <div>
      <lightning-input>
        #shadow-root(open)
      </lightning-input>
      <ul
        class="slds-m-top_x-small"
      >
        <li>
          <div
            class="slds-p-vertical_xx-small"
          >
            <p>
              Explore recipes
            </p>
            <p>
              Priority: 
              false
            </p>
          </div>
        </li>
        <li>
          <div
            class="slds-p-vertical_xx-small"
          >
            <p>
              Install Ebikes sample app
            </p>
            <p>
              Priority: 
              false
            </p>
          </div>
        </li>
      </ul>
    </div>
</c-todo-list>
`;

exports[`c-todo-list renders without any list items as default 1`] = `
<c-todo-list>
  #shadow-root(open)
    <div>
      <lightning-input>
        #shadow-root(open)
      </lightning-input>
      <ul
        class="slds-m-top_x-small"
      />
    </div>
</c-todo-list>
`;
